File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnDspFeature.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**************************************************************************
** GnDspFeature
*/
public class GnDspFeature : GnObject {
  private HandleRef swigCPtr;

  internal GnDspFeature(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnDspFeature_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnDspFeature obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnDspFeature() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnDspFeature(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public string FeatureData() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnDspFeature_FeatureData(swigCPtr);
    return ret;
  }

  public GnDspFeatureQuality FeatureQuality() {
    GnDspFeatureQuality ret = (GnDspFeatureQuality)gnsdk_csharp_marshalPINVOKE.GnDspFeature_FeatureQuality(swigCPtr);
    return ret;
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnDspFeature.java

package com.gracenote.gnsdk;

/************************************************************************** 
** {@link GnDspFeature} 
*/ 
 
public class GnDspFeature extends GnObject {
  private long swigCPtr;

  protected GnDspFeature(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnDspFeature_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnDspFeature obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnDspFeature(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public String featureData() {
    return gnsdk_javaJNI.GnDspFeature_featureData(swigCPtr, this);
  }

  public GnDspFeatureQuality featureQuality() {
    return GnDspFeatureQuality.swigToEnum(gnsdk_javaJNI.GnDspFeature_featureQuality(swigCPtr, this));
  }

}
